### PR TITLE
Added new option to select the active a10 partition

### PR DESCRIPTION
**Pull request by Babybolzer**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

a10
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```
##### SUMMARY

Added new option to select the active a10 partition. 

```
No new output
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @Babybolzer for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Babybolzer** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @Babybolzer. To the current maintainers, @mischapeters please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by mischapeters** [comment]
👍 shipit :shipit:

**Comment by gregdek** [comment]
Thanks again to @Babybolzer for this PR, and thanks @mischapeters for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: network/a10/a10_server.py [hunk 1]
```diff
@@ -96,6 +105,20 @@
 
 '''
 
+RETURN = '''
+msg:
```
Comment (resmo) [inline network/a10/a10_server.py]: this is a common return, please remove it from module description.

File: network/a10/a10_service_group.py [hunk 2]
```diff
@@ -123,6 +132,19 @@
 
 '''
 
+RETURN = '''
+msg:
```
Comment (resmo) [inline network/a10/a10_service_group.py]: this is a common return, please remove it from module description

File: network/a10/a10_virtual_server.py [hunk 3]
```diff
@@ -122,6 +131,19 @@
 
 '''
 
+RETURN = '''
+msg:
```
Comment (resmo) [inline network/a10/a10_virtual_server.py]: this is a common return, please remove it from module description

File: network/a10/a10_server.py [hunk 4]
```diff
@@ -46,6 +46,14 @@
       - admin password of your A10 Networks device
     required: true
     aliases: ['pass', 'pwd']
+  partition:
+    version_added: "2.2"
+    description:
+      - set active-partition
+    required: false
+    default: null
+    aliases: []
```
Comment (gundalow) [inline network/a10/a10_server.py]: Please remove empty options, this line (and similar isn't needed)

File: network/a10/a10_service_group.py [hunk 5]
```diff
@@ -174,6 +196,7 @@ def main():
                                                'src-ip-only-hash',
                                                'src-ip-hash']),
             servers=dict(type='list', aliases=['server', 'member'], default=[]),
+            partition=dict(type='str', aliases=['partition'], default=[]),
```
Comment (gundalow) [inline network/a10/a10_service_group.py]: `partition` is the name, so you can remove `aliases=['partition'],`

File: network/a10/a10_virtual_server.py [hunk 6]
```diff
@@ -170,6 +192,7 @@ def main():
             virtual_server_ip=dict(type='str', aliases=['ip', 'address'], required=True),
             virtual_server_status=dict(type='str', default='enabled', aliases=['status'], choices=['enabled', 'disabled']),
             virtual_server_ports=dict(type='list', required=True),
+            partition=dict(type='str', aliases=['partition'], default=[]),
```
Comment (gundalow) [inline network/a10/a10_virtual_server.py]: `partition` is the name, so you can remove `aliases=['partition'],`

File: network/a10/a10_server.py [hunk 7]
```diff
@@ -141,6 +164,7 @@ def main():
             server_ip=dict(type='str', aliases=['ip', 'address']),
             server_status=dict(type='str', default='enabled', aliases=['status'], choices=['enabled', 'disabled']),
             server_ports=dict(type='list', aliases=['port'], default=[]),
+            partition=dict(type='str', aliases=['partition'], default=[]),
```
Comment (gundalow) [inline network/a10/a10_server.py]: `partition` is the name, so you can remove `aliases=['partition'],`

**Comment by gundalow** [comment]
1) You don't need to document empty cases, so the following lines can be removed from all modules
`aliases: []`

`choices: []`

2) Does the module still work if you don't specify `partition`?

needs_revision

**Comment by gregdek** [comment]
Thanks @Babybolzer for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Babybolzer** [comment]
@gundalow thanks for the input. Yes, the module still works without specifying `partition`. The changes are then simply done to the default partition.

**Comment by Babybolzer** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @Babybolzer. To the current maintainers, @mischapeters please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @Babybolzer for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@Babybolzer A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Babybolzer** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @Babybolzer. To the current maintainers, @mischapeters please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by mischapeters** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @Babybolzer for this PR, and thanks @mischapeters for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @Babybolzer for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

File: network/a10/a10_server.py [hunk 8]
```diff
@@ -31,6 +31,26 @@
 author: "Mischa Peters (@mischapeters)"
 extends_documentation_fragment: a10
 options:
+  host:
```
Comment (gundalow) [inline network/a10/a10_server.py]: Hi,
You don't need to document host, username, or password as the are documented in https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/module_docs_fragments/a10.py

Which is pulled in by `extends_documentation_fragment: a10`

File: network/a10/a10_service_group.py [hunk 9]
```diff
@@ -33,6 +33,29 @@
     - When a server doesn't exist and is added to the service-group the server will be created
 extends_documentation_fragment: a10
 options:
+  host:
```
Comment (gundalow) [inline network/a10/a10_service_group.py]: Please remove host, username, and password.

File: network/a10/a10_service_group.py [hunk 10]
```diff
@@ -60,6 +83,22 @@
         specify the C(status:). See the examples below for details.
     required: false
     default: null
+  write_config:
```
Comment (gundalow) [inline network/a10/a10_service_group.py]: Please remove write_config and validate_certs

File: network/a10/a10_virtual_server.py [hunk 11]
```diff
@@ -31,6 +31,29 @@
 author: "Mischa Peters (@mischapeters)"
 extends_documentation_fragment: a10
 options:
+  host:
```
Comment (gundalow) [inline network/a10/a10_virtual_server.py]: Please remove host, username, and password.

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

File: network/a10/a10_service_group.py (hunk 10 above)
Comment (gundalow) [reply network/a10/a10_service_group.py]: Just checking you saw that `write_config` and `validate_certs` docs need removing as well

**Comment by Babybolzer** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @Babybolzer. To the current maintainers, @mischapeters, @ericchou1, @gundalow, @qalthos, @privateip please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Review by gundalow — APPROVED** [review]

**Pull request merged**